src/sentry/models/project.py

```diff
@@ -260,6 +260,43 @@ def get_full_name(self):
             return '%s %s' % (self.team.name, self.name)
         return self.name
 
+    def get_notification_recipients(self, user_option):
+        from sentry.models import UserOption
+        alert_settings = dict(
+            (o.user_id, int(o.value))
+            for o in UserOption.objects.filter(
+                project=self,
+                key=user_option,
+            )
+        )
+
+        disabled = set(u for u, v in six.iteritems(alert_settings) if v == 0)
+
+        member_set = set(self.member_set.exclude(
+            user__in=disabled,
+        ).values_list('user', flat=True))
+
+        # determine members default settings
+        members_to_check = set(u for u in member_set if u not in alert_settings)
+        if members_to_check:
+            disabled = set((
+                uo.user_id for uo in UserOption.objects.filter(
```

Comment:
This query needs to be qualified with a project. Otherwise, this doesn't hit an index.